File: Android/app/src/main/java/com/example/smartfashion/model/OutfitItem.kt
package com.example.smartfashion.model

import com.google.gson.annotations.SerializedName
import java.util.Date

data class OutfitItem(
    @SerializedName("item_id")
    val itemId: Int? = null,

    @SerializedName("outfit_id")
    val outfitId: Int,

    @SerializedName("item_ref_id")
    val itemRefId: Int,

    @SerializedName("item_type")
    val itemType: String = "personal",

    @SerializedName("position_x")
    val positionX: Double = 0.0,

    @SerializedName("position_y")
    val positionY: Double = 0.0,

    @SerializedName("scale")
    val scale: Double = 1.0,

    @SerializedName("rotation")
    val rotation: Double = 0.0,

    @SerializedName("z_index")
    val zIndex: Int = 0,

    @SerializedName("createdAt")
    val createdAt: Date? = null,

    @SerializedName("updatedAt")
    val updatedAt: Date? = null
)

File: Server/models/OutfitItem.js
const mongoose = require('mongoose');
const Counter = require('./Counter');

const outfitItemSchema = new mongoose.Schema({
  item_id: {
    type: Number,
    unique: true
  },
  outfit_id: {
    type: Number,
    required: true,
    ref: 'Outfit',
    index: true 
  },
  item_ref_id: {
    type: Number,
    required: true,
    index: true  
  },
  item_type: {
    type: String, 
    enum: ['personal', 'system'], 
    default: 'personal' 
  },
  position_x: {
    type: Number,
    required: true,
    default: 0
  },
  position_y: {
    type: Number,
    required: true,
    default: 0
  },
  scale: {
    type: Number,
    required: true,
    default: 1.0
  },
  rotation: {
    type: Number,
    required: true,
    default: 0
  },
  z_index: {
    type: Number, 
    required: true,
    default: 0
  }
}, {
  timestamps: true
});

outfitItemSchema.index({ outfit_id: 1, item_ref_id: 1, item_type: 1 });

outfitItemSchema.pre('save', async function() {
  if (!this.isNew) {
    return; 
  }

  const counter = await Counter.findByIdAndUpdate(
    { _id: 'item_id' },
    { $inc: { seq: 1 } },
    { new: true, upsert: true }
  );

  this.item_id = counter.seq;
});

module.exports = mongoose.model('OutfitItem', outfitItemSchema, 'outfit_items');